lib/plugins/aws/configCredentials/awsConfigCredentials.js

```diff
@@ -63,45 +77,93 @@ class AwsConfigCredentials {
-      if (credsFile.length && credsFile.indexOf(`[${this.options.profile}]`) > -1) {
+    // Check if the profile exists
+    const isNewProfile = profileBoundaries.start === -1;
+    if (isNewProfile) {
+      this.addProfile();
+    } else {
+      // Only update the profile if the overwrite flag was set
+      if (!this.options.overwrite) {
         this.serverless.cli.log(
-          `Failed! ~/.aws/credentials exists and already has a "${this.options.profile}" profile.`);
+          `Failed! ~/.aws/credentials already has a "${this.options.profile}" profile.
+          Use the overwrite flag ("-o" or "--overwrite") to force the update`);
         return BbPromise.resolve();
       }
-    } else {
-      // create the credentials file alongside the .aws directory if it's not yet present
-      fse.ensureFileSync(credsPath);
+
+      this.updateProfile(profileBoundaries);
     }
 
-    // write credentials file with 'default' profile
-    this.serverless.utils.appendFileSync(
-      credsPath,
-      `[${this.options.profile}]
-aws_access_key_id=${this.options.key}
-aws_secret_access_key=${this.options.secret}
-`); // Keep line break at the end. Otherwise will break AWS CLI.
+    this.saveCredentialsFile();
+    return BbPromise.resolve();
+  }
+
+  getCredentials() {
+    // Get the credentials file lines
+    const credentialsFileContent = this.serverless.utils.readFileSync(this.credentialsFilePath);
+    return credentialsFileContent ? credentialsFileContent.split('\n') : [];
+  }
+
+  addProfile() {
+    this.credentials.push(`[${this.options.profile}]`,
+      `aws_access_key_id = ${this.options.key}`,
+      `aws_secret_access_key = ${this.options.secret}`);
+  }
+
+  updateProfile(profileBoundries) {
+    let currentLine = profileBoundries.start;
+    let endLine = profileBoundries.end;
+
+    // Remove existing 'aws_access_key_id' and 'aws_secret_access_key' properties
+    while (currentLine < endLine) {
+      const line = this.credentials[currentLine];
+      if (
+        line.indexOf('aws_access_key_id') > -1 ||
+        line.indexOf('aws_secret_access_key') > -1
+      ) {
+        this.credentials.splice(currentLine, 1);
+        endLine--;
+      } else {
+        currentLine++;
+      }
+    }
+
+    // Add the key and the secret to the beginning of the section
+    const keyLine = `aws_access_key_id = ${this.options.key}`;
+    const secretLine = `aws_secret_access_key = ${this.options.secret}`;
+
+    this.credentials.splice(profileBoundries.start + 1, 0, secretLine);
+    this.credentials.splice(profileBoundries.start + 1, 0, keyLine);
+  }
+
+  saveCredentialsFile() {
+    // Generate the file content and add a line break at the end
+    const updatedCredsFileContent = `${this.credentials.join('\n')}\n`;
+
+    this.serverless.cli.log('Saving your AWS profile in "~/.aws/credentials"...');
+
+    this.serverless.utils.writeFile(this.credentialsFilePath, updatedCredsFileContent);
```

Comment:
this method is async, so the success message below and any other following operation would run regardless of the result of this writeFile. you will need a return to avoid race conditions:

```
return this.serverless.utils.writeFile(this.credentialsFilePath, updatedCredsFileContent).then(() => {
  this.serverless.cli.log(
        `Success! Your AWS access keys were stored under the "${this.options.profile}" profile.`);
});
```

Also, that would make the `saveCredentialsFile` return a promise, so that would affect whatever method is calling it the same way. We should be careful of the promise chain 😊